imagebuildah/executor.go

```diff
@@ -598,6 +598,25 @@ func (b *Executor) Build(ctx context.Context, stages imagebuilder.Stages) (image
 			for _, child := range node.Children { // tokens on this line, though we only care about the first
 				switch strings.ToUpper(child.Value) { // first token - instruction
 				case "FROM":
+					for _, flag := range child.Flags {
+						if strings.HasPrefix(flag, "--platform=") {
+							platformString := strings.TrimPrefix(flag, "--platform=")
```

Comment:
Would this be better to split on "=".
Is "--platform =" valid in a FROM?

Reply:
If not,
LGTM

Reply:
> Would this be better to split on "=".

@rhatdan Parsing is same as we parse other `flags` in executor's code. I am okay to change this one but my intention was to keep consistency for reader : https://github.com/containers/buildah/blob/main/imagebuildah/executor.go#L626


> Is "--platform =" valid in a FROM?

Yes `--platform` is a valid field in `docker` but its not supported by `buildah` following PR just adds support for that. Following issues states need for the same: https://github.com/containers/buildah/issues/3712 & https://github.com/containers/buildah/issues/3261#issuecomment-848838772